django_cas_ng/views.py

```diff
@@ -109,8 +109,7 @@ def login(request, next_page=None, required=False):
         elif settings.CAS_RETRY_LOGIN or required:
             return HttpResponseRedirect(_login_url(service))
         else:
-            error = "<h1>Forbidden</h1><p>Login failed.</p>"
-            return HttpResponseForbidden(error)
+            return HttpResponseRedirect(next_page)
```

Comment:
Given that this fix is specific to your use case, what do you think about adding a hook here to allow for custom logic, but keep the `HttpResponseForbidden` as the default.

Thoughts on how this could be implemented.

1.  A new setting.  Something like `CAS_ON_FAILURE_CALLBACK` which is a python path to a callable which returns an `HttpResponse`
2.  A new setting.  Something like `CAS_REDIRECT_ON_FAILURE` which returns the `HttpResponseRedirect` rather than  `HttpResponseForbidden`.

Or some other idea I haven't thought of.

I think I'm partial to number 2.